### PR TITLE
fix: refactor for updating dependencies in newspack-scripts

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Refactoring required for changes in https://github.com/Automattic/newspack-scripts/pull/209. Note that this PR will not pass CI tests and should NOT be merged until https://github.com/Automattic/newspack-scripts/pull/209 is released as an NPM package, so this repo can install it as a dependency.

### How to test the changes in this Pull Request:

Follow instructions in https://github.com/Automattic/newspack-scripts/pull/209.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,25 +5,24 @@
...
 		"cm": "newspack-scripts commit",
 		"semantic-release": "newspack-scripts release --files=newspack-newsletters.php",
-		"start": "npm ci && newspack-scripts watch",
-		"watch": "newspack-scripts watch",
-		"build": "newspack-scripts build",
-		"test": "echo \"Error: no test specified\" && exit 0",
+		"start": "npm ci && npm run watch",
+		"build": "newspack-scripts wp-scripts build",
+		"watch": "newspack-scripts wp-scripts start",
+		"test": "newspack-scripts test",
```
Comment (chickenn00dle) [inline package.json]: Running this, I get the following output:

```
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /Users/raz/Sites/newspack/newsletters
  397 files checked.
  testMatch: /Users/raz/Sites/newspack/newsletters/**/*test.js?(x) - 0 matches
  testPathIgnorePatterns: /node_modules/ - 397 matches
  testRegex:  - 0 matches
Pattern:  - 0 matches
```

Looks like there are tests in the repo though.

File: package.json [hunk 2]
```diff
@@ -5,25 +5,24 @@
...
 		"clean": "rm -rf dist/",
+		"lint": "npm run lint:scss && npm run lint:js",
+		"lint:js": "newspack-scripts wp-scripts lint-js '**/{src,includes}/**/*.{js,jsx,ts,tsx}'",
+		"lint:js:staged": "newspack-scripts wp-scripts lint-js --ext .js,.jsx,.ts,.tsx",
+		"format:js": "newspack-scripts wp-scripts format '**/{src,includes}/**/*.{js,jsx,ts,tsx}'",
```
Comment (chickenn00dle) [inline package.json:17]: Same question as in other repos: running this changes alot of files despite lint passing. Is this expected?

File: package.json [hunk 3]
```diff
@@ -5,25 +5,24 @@
...
-		"lint:scss:staged": "stylelint --customSyntax postcss-scss --config=./node_modules/newspack-scripts/config/stylelint.config.js",
-		"format:js": "prettier 'src/**/*.{js,jsx}' --write",
-		"format:scss": "prettier --write 'src/**/*.scss'",
-		"format:php": "./vendor/bin/phpcbf .",
-		"release:archive": "rm -rf release && mkdir -p release && rsync -r . ./release/newspack-newsletters --exclude-from='./.distignore' && cd release && zip -r newspack-newsletters.zip newspack-newsletters",
-		"release": "npm run build && npm run semantic-release"
+		"lint:scss": "newspack-scripts wp-scripts lint-style '**/{src,includes}/**/*.scss' --customSyntax postcss-scss",
+		"lint:scss:staged": "newspack-scripts wp-scripts lint-style --customSyntax postcss-scss",
+		"format:scss": "newspack-scripts wp-scripts lint-style '**/{src,includes}/**/*.scss' --customSyntax postcss-scss --fix",
+		"typescript:check": "newspack-scripts typescript-check",
```
Comment (chickenn00dle) [inline package.json]: tcs help is included in the output when running this. Not sure if this is expected or if perhaps the command is being called incorrectly by the script for some reason?

Screenshot 2024-06-20 at 16 55 46

**Review by chickenn00dle — DISMISSED** [review]
Haven't gotten the chance to smoke test this one yet, but will go ahead and submit a first round of review since I'm ending my day soon and ran into some issues with the `npm` commands

File: package.json (hunk 1 above)
Comment (dkoo) [reply package.json]: Ah, there are no JS Jest tests in this repo. We can probably remove this script from repos that lack them, or change them to simply output that there are no tests implemented (yet).

File: package.json (hunk 3 above)
Comment (dkoo) [reply package.json]: Same issue as with Jest tests—there's no Typescript in this repo. We can omit or update the script for repos that lack any Typescript files.

**Review by miguelpeixe — APPROVED** [review]

**Pull request merged**